src/musicbrainz/musicbrainzclient.cpp

```diff
@@ -90,21 +91,27 @@ void MusicBrainzClient::requestFinished() {
     ResultList ret;
 
     int status = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
-    qDebug() << "MusicBrainzClient GET reply status:" << status;
+    QTextStream textReader(reply);
+    const QByteArray body(reply->readAll());
+    QXmlStreamReader reader(body);
 
     // MusicBrainz returns 404 when the MBID is not in their database. We treat
     // a status of 404 the same as a 200 but it will produce an empty list of
     // results.
     if (status != 200 && status != 404) {
+        qDebug() << "MusicBrainzClient POST reply status:" << status << "body:" << body;
+        QJsonDocument jsonResponse = QJsonDocument::fromJson(body);
+        QJsonObject jsonObject = jsonResponse.object();
+        QString message = jsonObject["error"].toString();
+        QStringList propertyNames;
+        QStringList propertyKeys;
+        QString strReply = (QString)reply->readAll();
         emit(networkError(
              reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt(),
-             "MusicBrainz"));
+             "MusicBrainz", message, 0));
```

Comment:
Passing 0 is not intuitive. Does a default parameter for signals work in Qt5? Otherwise use a constant with a concise name.